03-explanation/01-foundations/automation-paradigms/single-step-commands.mdx

```diff
@@ -0,0 +1,61 @@
+---
+title: "Single-Step Commands"
+description: "Use explicit commands when agentic instructions aren't reliable enough"
+---
+
+Single-step commands serve as an escape hatch when agentic instructions cannot handle specific automation cases reliably enough. While AI models are still involved in understanding visual elements and executing commands, this paradigm provides explicit control over the automation workflow.
+
+## Philosophy of Single-Step Commands
+
+Single-step commands operate on the principle of **explicit control**, where every action is deliberately specified by the developer. This approach provides:
+
+- **Fine-grained control**: Each command specifies exactly what action to take
+- **Higher reproducibility**: More consistent behavior across runs (though not deterministic)
+- **Debugging clarity**: Easy to identify exactly where issues occur
+- **Granular specification**: Precise control over individual interactions
+
+```python
+# Single-step approach: explicit control
+with VisionAgent() as agent:
+    agent.click("username field")
+    agent.type("username field", "john.doe")
+    agent.click("password field")
+    agent.type("password field", "secret123")
+    agent.click("Submit button")
+```
+
+
+## Core Components
+
+### Element Selection
+
+Single-step commands rely on precise element selection methods:
+
+- **Natural Language Selection**: Describe elements in plain English ("submit button", "username field")
```

Comment:
The default model is an OCR model. 

If you want to use plain english you need to sue `loc.Prompt("submit button")`